src/components/v2/V2Project/V2ConfirmPayOwnerModal/index.tsx

```diff
@@ -0,0 +1,244 @@
+import FormattedAddress from 'components/shared/FormattedAddress'
+import ImageUploader from 'components/shared/inputs/ImageUploader'
+import { NetworkContext } from 'contexts/networkContext'
+import { useCurrencyConverter } from 'hooks/v1/CurrencyConverter'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import React, { useContext, useState } from 'react'
+import { formattedNum, formatWad, fromWad } from 'utils/formatNumber'
+
+import { tokenSymbolText } from 'utils/tokenSymbolText'
+import {
+  V2CurrencyName,
+  V2_CURRENCY_ETH,
+  V2_CURRENCY_USD,
+} from 'utils/v2/currency'
+import { FormItems } from 'components/shared/formItems'
+import { PayV2ProjectTxType } from 'hooks/v2/transactor/PayV2ProjectTx'
+import { ethers, utils } from 'ethers'
+import { weightedRate } from 'utils/math'
+
+import V2ProjectRiskNotice from './V2ProjectRiskNotice'
+
+export default function V2ConfirmPayOwnerModal({
+  visible,
+  weiAmount,
+  onSuccess,
+  onCancel,
+  payProjectTx,
+}: {
+  visible?: boolean
+  weiAmount: BigNumber | undefined
+  onSuccess?: VoidFunction
+  onCancel?: VoidFunction
+  payProjectTx: PayV2ProjectTxType
+}) {
+  const { userAddress } = useContext(NetworkContext)
+
+  const [loading, setLoading] = useState<boolean>()
+  const [preferClaimed, setPreferClaimed] = useState<boolean>(false)
+  const [customBeneficiaryEnabled, setCustomBeneficiaryEnabled] =
+    useState<boolean>(false)
+  const [beneficiary, setBeneficiary] = useState<string | undefined>(
+    userAddress,
+  )
+
+  const [form] = useForm<{ memo: string; beneficiary: string }>()
+  const { fundingCycle, projectMetadata } = useContext(V2ProjectContext)
+  const converter = useCurrencyConverter()
+
+  const usdAmount = converter.weiToUsd(weiAmount)
+
+  async function pay() {
+    if (!weiAmount) return
+    await form.validateFields()
+
+    if (userAddress) {
+      setLoading(true)
+    }
+
+    payProjectTx(
+      {
+        memo: form.getFieldValue('memo'),
+        preferClaimedTokens: preferClaimed,
+        beneficiary: beneficiary,
+        value: weiAmount, //'0.001'),//BigNumber.from(0.01)//weiAmount,
+      },
+      {
+        onConfirmed: () => {
+          if (onSuccess) onSuccess()
+        },
+        onDone: () => setLoading(false),
+      },
+    )
+  }
+
+  const validateCustomBeneficiary = () => {
+    if (!beneficiary) {
+      return Promise.reject('Address required')
+    } else if (!utils.isAddress(beneficiary)) {
+      return Promise.reject('Invalid address')
+    }
+    return Promise.resolve()
+  }
+
+  // TODO: reserved rate hardcoded to 0
+  const receivedTickets = weightedRate(
+    fundingCycle?.weight,
+    0,
+    weiAmount,
+    'payer',
+  ) // TODO: reserved rate hardcoded to 0
+  const ownerTickets = weightedRate(
+    fundingCycle?.weight,
+    0,
+    weiAmount,
+    'reserved',
+  )
+
+  const hasIssuedTokens = true //tokenAddress && tokenAddress !== constants.AddressZero
```

Comment:
**question**: can we get `tokenAddress` somehow? Might need a hook `ProjectToken`, that calls `tokenOf(projectId)` on the `JBTokenStore` contract:

https://docs.juicebox.money/protocol/specifications/contracts/jbtokenstore/properties/tokenof

Reply:
Yeah cool done ✅

Reply:
flippen awesome